src/main.rs

```diff
@@ -1453,8 +1453,20 @@ mod tests {
         test_block!(tb, "", {
             let mut clear = ClearDiskConfig::new();
             let guest = Guest::new(&mut clear);
+            let sock_path = String::from(guest.tmp_dir.path().join("vub.sock").to_str().unwrap());
+            let mut workload_path = dirs::home_dir().unwrap();
+            workload_path.push("workloads");
+            workload_path.push("blk.img");
+            let image_path = workload_path.to_str().unwrap();
 
-            let (mut daemon_child, vubd_socket_path) = prepare_vubd(&guest.tmp_dir, "blk.img");
+            // Start the daemon
+            let mut daemon_child = Command::new("target/debug/vhost_user_blk")
+                .args(&[
+                    "--backend",
+                    format!("image={},sock={}", image_path, sock_path).as_str(),
+                ])
+                .spawn()
+                .unwrap();
```

Comment:
Instead of adding all this code, could you instead modify the `prepare_vubd()` function so that it spawns the `vhost-user-blk` daemon, using your new binary. This way the necessary changes should even be less intrusive.